ledger/src/snapshot_utils.rs

```diff
@@ -547,6 +547,62 @@ pub fn get_highest_snapshot_archive_path<P: AsRef<Path>>(
     archives.into_iter().next()
 }
 
+const MAXIMUM_SNAPSHOT_ARCHIVE_SIZE: u64 = 500 * 1024 * 1024 * 1024; // 500 GiB
+
+fn is_valid_archive_entry(parts: &[&str], kind: tar::EntryType) -> bool {
+    use tar::EntryType::{Directory, GNUSparse, Regular};
+    let storage_regex = Regex::new(r"^\d+.\d+$").unwrap();
+    let slot_regex = Regex::new(r"^\d+$").unwrap();
+
+    trace!("validating: {:?} {:?}", parts, kind);
+    match (parts, kind) {
+        (["version"], Regular) => true,
```

Comment:
Currently any symlinks are allowed to be unpacked... https://github.com/solana-labs/solana/issues/7167#issuecomment-599479747